[EngSys] TS5055: Cannot write file 'index.d.ts' because it would overwrite input file.

If you build the repo multiple times, you might get this error:

```
npm error error TS5055: Cannot write file 'eng/tools/typespec-validation/dist/src/index.d.ts' because it would overwrite input file.
```

I suspect this is a small bug in our tsconfig files.

--- a/eng/tools/lint-diff/tsconfig.json
+++ b/eng/tools/lint-diff/tsconfig.json
@@ -2,6 +2,7 @@
   "extends": "../tsconfig.json",
   "compilerOptions": {
     "outDir": "./dist",
+    "rootDir": ".",
     "noImplicitReturns": true,
     "allowJs": true,
   },

--- a/eng/tools/lint-diff/tsconfig.json
+++ b/eng/tools/lint-diff/tsconfig.json
@@ -7,6 +7,7 @@
     "allowJs": true,
   },
   "include": [
+    "*.ts",
     "src/**/*.ts",
     "test/**/*.ts",
   ],

--- a/eng/tools/lint-diff/tsconfig.json
+++ b/eng/tools/lint-diff/tsconfig.json
@@ -5,4 +5,8 @@
     "noImplicitReturns": true,
     "allowJs": true,
   },
+  "include": [
+    "src/**/*.ts",
+    "test/**/*.ts",
+  ],
 }

--- a/eng/tools/sdk-suppressions/tsconfig.json
+++ b/eng/tools/sdk-suppressions/tsconfig.json
@@ -3,5 +3,9 @@
   "compilerOptions": {
     "outDir": "./dist",
     "allowJs": true
-  }
+  },
+  "include": [
+    "src/**/*.ts",
+    "test/**/*.ts",
+  ],
 }

--- a/eng/tools/sdk-suppressions/tsconfig.json
+++ b/eng/tools/sdk-suppressions/tsconfig.json
@@ -2,6 +2,7 @@
   "extends": "../tsconfig.json",
   "compilerOptions": {
     "outDir": "./dist",
+    "rootDir": ".",
     "allowJs": true
   },
   "include": [

--- a/eng/tools/sdk-suppressions/tsconfig.json
+++ b/eng/tools/sdk-suppressions/tsconfig.json
@@ -6,6 +6,7 @@
     "allowJs": true
   },
   "include": [
+    "*.ts",
     "src/**/*.ts",
     "test/**/*.ts",
   ],

--- a/eng/tools/spec-gen-sdk-runner/tsconfig.json
+++ b/eng/tools/spec-gen-sdk-runner/tsconfig.json
@@ -5,5 +5,9 @@
     "module": "ESNext",
     "target": "ESNext",
     "moduleResolution": "node",
-  }
+  },
+  "include": [
+    "src/**/*.ts",
+    "test/**/*.ts",
+  ],
 }

--- a/eng/tools/spec-gen-sdk-runner/tsconfig.json
+++ b/eng/tools/spec-gen-sdk-runner/tsconfig.json
@@ -2,6 +2,7 @@
   "extends": "../tsconfig.json",
   "compilerOptions": {
     "outDir": "./dist",
+    "rootDir": ".",
     "module": "ESNext",
     "target": "ESNext",
     "moduleResolution": "node",

--- a/eng/tools/spec-gen-sdk-runner/tsconfig.json
+++ b/eng/tools/spec-gen-sdk-runner/tsconfig.json
@@ -8,6 +8,7 @@
     "moduleResolution": "node",
   },
   "include": [
+    "*.ts",
     "src/**/*.ts",
     "test/**/*.ts",
   ],

--- a/eng/tools/specs-model/tsconfig.json
+++ b/eng/tools/specs-model/tsconfig.json
@@ -6,6 +6,7 @@
   ],
   "compilerOptions": {
     "outDir": "./dist",
+    "rootDir": ".",
     "target": "ES2022",
     "lib": ["ES2022"],
 

--- a/eng/tools/specs-model/tsconfig.json
+++ b/eng/tools/specs-model/tsconfig.json
@@ -16,6 +16,7 @@
     "checkJs": false
   },
   "include": [
+    "*.ts",
     "src/**/*.ts",
     "test/**/*.ts",
   ],

--- a/eng/tools/specs-model/tsconfig.json
+++ b/eng/tools/specs-model/tsconfig.json
@@ -13,7 +13,11 @@
     // We don't need it anyway, as all sources are in .ts except the 3-line cmd entry-point.
     // https://www.typescriptlang.org/tsconfig/#checkJs
     "checkJs": false
-  }
+  },
+  "include": [
+    "src/**/*.ts",
+    "test/**/*.ts",
+  ],
 }
 
 // [strictest]: https://www.npmjs.com/package/@tsconfig/strictest from [tsconfig bases]

--- a/eng/tools/suppressions/tsconfig.json
+++ b/eng/tools/suppressions/tsconfig.json
@@ -5,6 +5,7 @@
     "rootDir": ".",
   },
   "include": [
+    "*.ts",
     "src/**/*.ts",
     "test/**/*.ts",
   ],

--- a/eng/tools/suppressions/tsconfig.json
+++ b/eng/tools/suppressions/tsconfig.json
@@ -2,5 +2,9 @@
   "extends": "../tsconfig.json",
   "compilerOptions": {
     "outDir": "./dist",
-  }
+  },
+  "include": [
+    "src/**/*.ts",
+    "test/**/*.ts",
+  ],
 }

--- a/eng/tools/suppressions/tsconfig.json
+++ b/eng/tools/suppressions/tsconfig.json
@@ -2,6 +2,7 @@
   "extends": "../tsconfig.json",
   "compilerOptions": {
     "outDir": "./dist",
+    "rootDir": ".",
   },
   "include": [
     "src/**/*.ts",

--- a/eng/tools/tsconfig.json
+++ b/eng/tools/tsconfig.json
@@ -10,6 +10,8 @@
     // required to use project references
     "composite": true,
   },
+  // Compile nothing at this level
+  "files": [],
   "references": [
     { "path": "./specs-model" },
     { "path": "./suppressions" },

--- a/eng/tools/tsp-client-tests/tsconfig.json
+++ b/eng/tools/tsp-client-tests/tsconfig.json
@@ -5,6 +5,7 @@
     "rootDir": ".",
   },
   "include": [
+    "*.ts",
     "src/**/*.ts",
     "test/**/*.ts",
   ],

--- a/eng/tools/tsp-client-tests/tsconfig.json
+++ b/eng/tools/tsp-client-tests/tsconfig.json
@@ -2,5 +2,9 @@
   "extends": "../tsconfig.json",
   "compilerOptions": {
     "outDir": "./dist",
-  }
+  },
+  "include": [
+    "src/**/*.ts",
+    "test/**/*.ts",
+  ],
 }

--- a/eng/tools/tsp-client-tests/tsconfig.json
+++ b/eng/tools/tsp-client-tests/tsconfig.json
@@ -2,6 +2,7 @@
   "extends": "../tsconfig.json",
   "compilerOptions": {
     "outDir": "./dist",
+    "rootDir": ".",
   },
   "include": [
     "src/**/*.ts",

--- a/eng/tools/typespec-requirement/tsconfig.json
+++ b/eng/tools/typespec-requirement/tsconfig.json
@@ -5,6 +5,7 @@
     "rootDir": ".",
   },
   "include": [
+    "*.ts",
     "src/**/*.ts",
     "test/**/*.ts",
   ],

--- a/eng/tools/typespec-requirement/tsconfig.json
+++ b/eng/tools/typespec-requirement/tsconfig.json
@@ -2,5 +2,9 @@
   "extends": "../tsconfig.json",
   "compilerOptions": {
     "outDir": "./dist",
-  }
+  },
+  "include": [
+    "src/**/*.ts",
+    "test/**/*.ts",
+  ],
 }

--- a/eng/tools/typespec-requirement/tsconfig.json
+++ b/eng/tools/typespec-requirement/tsconfig.json
@@ -2,6 +2,7 @@
   "extends": "../tsconfig.json",
   "compilerOptions": {
     "outDir": "./dist",
+    "rootDir": ".",
   },
   "include": [
     "src/**/*.ts",

--- a/eng/tools/typespec-validation/tsconfig.json
+++ b/eng/tools/typespec-validation/tsconfig.json
@@ -2,6 +2,7 @@
   "extends": "../tsconfig.json",
   "compilerOptions": {
     "outDir": "./dist",
+    "rootDir": ".",
     "allowJs": true
   },
   "include": [

--- a/eng/tools/typespec-validation/tsconfig.json
+++ b/eng/tools/typespec-validation/tsconfig.json
@@ -6,6 +6,7 @@
     "allowJs": true
   },
   "include": [
+    "*.ts",
     "src/**/*.ts",
     "test/**/*.ts",
   ],

--- a/eng/tools/typespec-validation/tsconfig.json
+++ b/eng/tools/typespec-validation/tsconfig.json
@@ -4,6 +4,10 @@
     "outDir": "./dist",
     "allowJs": true
   },
+  "include": [
+    "src/**/*.ts",
+    "test/**/*.ts",
+  ],
   "references": [
     { "path": "../suppressions" }
   ]

